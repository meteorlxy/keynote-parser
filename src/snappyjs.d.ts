declare module 'snappyjs' {
  export function compress(input: Buffer): Buffer;
  export function uncompress(input: Buffer): Buffer;
}
