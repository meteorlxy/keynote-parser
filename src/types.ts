import type proto from '#proto/proto';

export interface IwaData {
  [archiveInfoIdentifier: string]: IwaArchiveInfoData;
}

export interface IwaArchiveInfoData {
  archiveInfo: proto.TSP.ArchiveInfo;
  messages: IwaMessageInfoData[];
}

export interface IwaMessageInfoData<T = unknown> {
  messageInfo: proto.TSP.IMessageInfo;
  messageProtoName: string;
  message: T;
}
