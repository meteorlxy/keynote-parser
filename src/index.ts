export * from './parse';
export type * from './types';
export * from './utils';
