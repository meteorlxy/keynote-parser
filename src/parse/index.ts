export * from './parse-iwa';
export * from './parse-iwa-protobuf';
export * from './parse-iwa-protobuf-archive-info';
export * from './parse-iwa-snappy';
export * from './parse-iwa-snappy-frame';
