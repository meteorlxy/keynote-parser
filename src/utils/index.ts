export * from './get-message-proto';
