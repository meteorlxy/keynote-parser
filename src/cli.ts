import path from 'node:path';
import process from 'node:process';
import { parse } from './parse';

export const cli = (): void => {
  // get input and output args
  const input = process.argv[2];
  const output = process.argv[3] || `${input}.parsed`;

  // resolve paths according to cwd
  const cwd = process.cwd();
  const inputPath = path.resolve(cwd, input);
  const outputPath = path.resolve(cwd, output);

  // parse
  parse(inputPath, outputPath).catch((error) => {
    console.error(error);
    process.exit(1);
  });
};
