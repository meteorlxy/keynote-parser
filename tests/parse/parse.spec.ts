import path from 'node:path';
import fastGlob from 'fast-glob';
import fs from 'fs-extra';
import { describe, expect, it } from 'vitest';
import { parse } from '../../src';

const KEYNOTE_FILE = path.resolve(__dirname, '../__fixtures__/empty.key');
const KEYNOTE_PARSED_DIR = path.resolve(__dirname, '../__temps__/empty');

describe('parse', () => {
  it('should work well', async () => {
    await fs.remove(KEYNOTE_PARSED_DIR);
    await parse(KEYNOTE_FILE, KEYNOTE_PARSED_DIR);
    const parsedFiles = await fastGlob('**', {
      cwd: KEYNOTE_PARSED_DIR,
      absolute: false,
    });
    expect(parsedFiles.sort()).toMatchSnapshot();
  });
});
