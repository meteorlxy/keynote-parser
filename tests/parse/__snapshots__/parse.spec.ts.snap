// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse > should work well 1`] = `
[
  "Data/660384004_1290x1720-small-5876.jpeg",
  "Data/660384004_rotate_1964x1473-small-8049.jpeg",
  "Data/666699290_02_crop_3159x1892-small-7420.jpeg",
  "Data/886640052_1858x1486-small-8023.jpeg",
  "Data/886640052_3195x2556-small-6274.jpeg",
  "Data/910457886_1434x1669-small-5644.jpeg",
  "Data/mt-1D8F67E0-B1C8-41A5-B712-33694CEDE772-8955.jpg",
  "Data/mt-3245D7CB-AABC-420F-B5DF-CF20E09F2549-8957.jpg",
  "Data/mt-64C85A4E-A48F-44BB-BE5C-EEC05B7245BF-8947.jpg",
  "Data/mt-6CA61296-A31B-4AD7-8EAA-035ADE58CEA1-8953.jpg",
  "Data/mt-71B3645B-721A-4CF9-BF24-FF1983BEA87B-8958.jpg",
  "Data/mt-786F1B80-948C-454C-911D-60B7B9889C85-8948.jpg",
  "Data/mt-9217E1E0-CC64-4424-8801-9370A8335C4A-8954.jpg",
  "Data/mt-9E6BBDBC-ED12-4A56-B16A-955A6B4EC923-8950.jpg",
  "Data/mt-B0B496FE-5CA3-4EF1-BF74-B1F15EADB01E-8961.jpg",
  "Data/mt-B863906C-3C09-43F8-8873-99CCCCD5A74F-8951.jpg",
  "Data/mt-C1F60A89-C497-4BFA-ABFE-1FA58788FA16-8949.jpg",
  "Data/mt-C6CF2C2C-5E51-4969-A9EC-B38A08EB06B5-8952.jpg",
  "Data/mt-E95A17B4-AE85-4C43-9F84-B6D856A7138C-8960.jpg",
  "Data/mt-ED5E8464-FA5C-49FD-BD0A-FFC6CCE2CA94-8959.jpg",
  "Data/mt-FB8D87C6-D53E-4208-ACBE-BB17F068AAF6-8956.jpg",
  "Data/st-30E040F1-3986-4526-BFE2-ED2D87B955A4-212.jpg",
  "Index/AnnotationAuthorStorage.iwa.json",
  "Index/CalculationEngine.iwa.json",
  "Index/Document.iwa.json",
  "Index/DocumentMetadata.iwa.json",
  "Index/DocumentStylesheet.iwa.json",
  "Index/Metadata.iwa.json",
  "Index/Slide-2644420.iwa.json",
  "Index/TemplateSlide-2644082.iwa.json",
  "Index/TemplateSlide-2644110.iwa.json",
  "Index/TemplateSlide-2644136.iwa.json",
  "Index/TemplateSlide-2644158.iwa.json",
  "Index/TemplateSlide-2644181.iwa.json",
  "Index/TemplateSlide-2644200.iwa.json",
  "Index/TemplateSlide-2644226.iwa.json",
  "Index/TemplateSlide-2644244.iwa.json",
  "Index/TemplateSlide-2644266.iwa.json",
  "Index/TemplateSlide-2644288.iwa.json",
  "Index/TemplateSlide-2644306.iwa.json",
  "Index/TemplateSlide-2644328.iwa.json",
  "Index/TemplateSlide-2644350.iwa.json",
  "Index/TemplateSlide-2644380.iwa.json",
  "Index/TemplateSlide-2644402.iwa.json",
  "Index/ViewState.iwa.json",
  "Metadata/BuildVersionHistory.plist",
  "Metadata/DocumentIdentifier",
  "Metadata/Properties.plist",
  "preview-micro.jpg",
  "preview-web.jpg",
  "preview.jpg",
]
`;
